### PR TITLE
Static analysis improvements python

### PR DESCRIPTION
**Pull request by datumbox**
- [x] Simplify expressions and logic
- [x] Remove unused vars.
- [x] Remove unnecessary parenthesis.

My IDE gets all the credit. :D

### CODE REVIEW

**Review by vfdev-5 — APPROVED** [review]
Thanks @datumbox. Looks good !

File: torchvision/models/detection/backbone_utils.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def resnet_fpn_backbone(
         norm_layer=norm_layer)
 
     # select layers that wont be frozen
-    assert trainable_layers <= 5 and trainable_layers >= 0
+    assert 5 >= trainable_layers >= 0
```
Comment (zhiqwang) [inline torchvision/models/detection/backbone_utils.py:90]: Haha, maybe `0 <= trainable_layers <=5` is more readable?

Comment (datumbox) [reply torchvision/models/detection/backbone_utils.py:90]: Great feedback for my IDE. Let me change this. :D

File: torchvision/ops/feature_pyramid_network.py [hunk 2]
```diff
@@ -99,9 +99,7 @@ def get_result_from_inner_blocks(self, x: Tensor, idx: int) -> Tensor:
         This is equivalent to self.inner_blocks[idx](x),
         but torchscript doesn't support this yet
         """
-        num_blocks = 0
-        for m in self.inner_blocks:
-            num_blocks += 1
+        num_blocks = len(self.inner_blocks)
```
Comment (fmassa) [inline torchvision/ops/feature_pyramid_network.py:102]: Oh, great to see that this is already supported by torchscript :-)

**Pull request merged**